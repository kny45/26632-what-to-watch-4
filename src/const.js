export const MovieScore = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  VERRY_GOOD: 10
};

export const MovieLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERRY_GOOD: `Verry good`,
  AWESOME: `Awesome`
};

export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const GENRE_ALL = `All genres`;

export const SHOWED_ITEMS_IN_MOVIES_LIST_DEFAULT = 8;

export const TIME_OUT = 1000;

export const VideoPlayerMode = {
  SMALL_MOVIE_CARD: `small movie card`,
  BIG_MOVIE_PLAYER: `big movie player`,
};
