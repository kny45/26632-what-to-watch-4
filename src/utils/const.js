export const MovieScore = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  VERRY_GOOD: 10
};

export const MovieLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERRY_GOOD: `Verry good`,
  AWESOME: `Awesome`
};

export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const GENRE_ALL = `All genres`;

export const SHOWED_ITEMS_IN_MOVIES_LIST_DEFAULT = 8;

export const TIME_OUT = 1000;

export const VideoPlayerMode = {
  SMALL_MOVIE_CARD: `small movie card`,
  BIG_MOVIE_PLAYER: `big movie player`,
};

export const AppPage = {
  MAIN_PAGE: `main screen`,
  MOVIE_PAGE: `movie screen`,
  BIG_MOVIE_PLAYER: `big movie player`,
  SIGN_IN: `sign in screen`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  MAIN: `/`,
  SING_IN: `/sign-in`,
  MY_LIST: `/mylist`,
  FILM: `/films/:id`,
  ADD_REVIEW: `/films/:id/review`,
};
