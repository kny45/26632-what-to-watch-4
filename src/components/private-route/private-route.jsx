import React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {AppRoute} from "../../utils/const.js";
import {AuthorizationStatus} from "../../utils/const.js";
import {
  getAuthorizationStatus,
  getAuthorizationStatusLoaded,
} from "../../reducer/user/selector.js";
import Loader from "../loader/loader.jsx";

const PrivateRoute = (props) => {
  const {
    render,
    path,
    exact,
    authorizationStatus,
    authorizationStatusLoaded,
  } = props;

  return (
    <Route
      path={path}
      exact={exact}
      render={(routeProps) => {
        if (authorizationStatusLoaded) {
          return (
            authorizationStatus === AuthorizationStatus.AUTH
              ? render(routeProps)
              : <Redirect to={AppRoute.LOGIN} />
          );
        }

        return <Loader />;
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  authorizationStatusLoaded: PropTypes.bool.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  authorizationStatusLoaded: getAuthorizationStatusLoaded(state),
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
