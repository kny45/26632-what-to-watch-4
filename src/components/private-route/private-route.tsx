import * as React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {AppRoute} from "../../utils/const";
import {AuthorizationStatus} from "../../utils/const";
import {
  getAuthorizationStatus,
  getAuthorizationStatusLoaded,
} from "../../reducer/user/selector";
import Loader from "../loader/loader";

const PrivateRoute = (props) => {
  const {
    render,
    path,
    exact,
    authorizationStatus,
    authorizationStatusLoaded,
  } = props;

  return (
    <Route
      path={path}
      exact={exact}
      render={(routeProps) => {
        if (authorizationStatusLoaded) {
          return (
            authorizationStatus === AuthorizationStatus.AUTH
              ? render(routeProps)
              : <Redirect to={AppRoute.LOGIN} />
          );
        }

        return <Loader />;
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  authorizationStatusLoaded: PropTypes.bool.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  authorizationStatusLoaded: getAuthorizationStatusLoaded(state),
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
