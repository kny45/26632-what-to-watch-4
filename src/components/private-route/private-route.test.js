import React from "react";
import {Router} from "react-router-dom";
import {Provider} from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {PrivateRoute} from "./private-route.jsx";

import {AuthorizationStatus} from "../../utils/const.js";
import history from "../../history.js";

const mockStore = configureStore([]);
const MockComponent = () => <div />;

describe(`Render PrivateRoute`, () => {
  const store = mockStore({});

  it(`PrivateRoute render with AuthorizationStatus.AUTH`, () => {
    const tree = renderer
      .create(
          <Provider store = {store}>
            <Router history = {history} >
              <PrivateRoute
                path = {`/`}
                exact = {true}
                authorizationStatus = {AuthorizationStatus.AUTH}
                render = {() => {
                  return (
                    <MockComponent />
                  );
                }}
              />
            </Router>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`PrivateRoute render with AuthorizationStatus.AUTH`, () => {
    const tree = renderer
      .create(
          <Provider store = {store}>
            <Router history = {history} >
              <PrivateRoute
                path = {`/`}
                exact = {true}
                authorizationStatus = {AuthorizationStatus.AUTH}
                render = {() => {
                  return (
                    <MockComponent />
                  );
                }}
              />
            </Router>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

