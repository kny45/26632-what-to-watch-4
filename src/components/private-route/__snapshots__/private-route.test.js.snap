// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PrivateRoute PrivateRoute render with AuthorizationStatus.AUTH 1`] = `<div />`;

exports[`Render PrivateRoute PrivateRoute render with AuthorizationStatus.AUTH 2`] = `<div />`;
